src/main/java/org/entur/lamassu/model/id/DefaultIdValidator.java

```diff
@@ -0,0 +1,123 @@
+/*
+ *
+ *
+ *  * Licensed under the EUPL, Version 1.2 or – as soon they will be approved by
+ *  * the European Commission - subsequent versions of the EUPL (the "Licence");
+ *  * You may not use this work except in compliance with the Licence.
+ *  * You may obtain a copy of the Licence at:
+ *  *
+ *  *   https://joinup.ec.europa.eu/software/page/eupl
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the Licence is distributed on an "AS IS" basis,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the Licence for the specific language governing permissions and
+ *  * limitations under the Licence.
+ *
+ */
+
+package org.entur.lamassu.model.id;
+
+public class DefaultIdValidator implements IdValidator {
+
+  public static final char ID_SEPARATOR_CHAR = ':';
+  public static final int ID_CODESPACE_LENGTH = 3;
+  public static final int ID_MINIMUM_LENGTH = 6;
+
+  protected static final DefaultIdValidator instance = new DefaultIdValidator();
+
+  public static DefaultIdValidator getInstance() {
+    return instance;
+  }
+
+  public boolean validate(CharSequence string, int offset, int length) {
+    // minimum size is XXX:X:X
+    if (length < ID_MINIMUM_LENGTH) {
+      return false;
+    }
+    if (string.charAt(offset + ID_CODESPACE_LENGTH) != ':') {
+      return false;
+    }
+    int last = getLastSeperatorIndex(string, ID_CODESPACE_LENGTH + 1, length);
```

Comment:
This explicitly disallows colon as a an ID character of ingested feeds, no? Shouldn't it just be the next colon instead of the last, if the substring between `ID_CODESPACE_LENGTH+1` and `last` (IMHO `next`) is expected to be the type?

Reply:
https://github.com/entur/lamassu/pull/516/commits/a29fe61799fe88a0832e2edf07ea73589c1a9f22